cepheus-cep/src/test/java/com/orange/cepheus/cep/SubscriptionManagerTest.java

```diff
@@ -235,4 +251,16 @@ private  void callSuccessCallback (ArgumentCaptor<SuccessCallback> successArg) {
         response.setSubscribeResponse(subscribeResponse);
         successArg.getValue().onSuccess(response);
     }
+
+    //Setting Provider Fiware ServiceName and Fiware ServicePath
+    @Test
+    public void testGetHeadersForProvider() {
+        
+        Provider provider = new Provider();
+        provider.setUrl("http://localhost:1026/");
+        provider.setServiceName("Abc");
+        provider.setServicePath("/");
+        when(ngsiClient.getRequestHeaders(provider.getUrl())).thenReturn(new HttpHeaders());
```

Comment:
This is not testing the proposed change. A proper test would check that a subscription is made using the service name/path provided in the configuration by the provider.

It shoud look like a modified version of the`setConfigurationOK` test.